src/internal/m365/collection/drive/delta_tree.go

```diff
@@ -336,32 +335,33 @@ func (face *folderyMcFolderFace) addFile(
 		parentID     string
 	)
 
+	if len(id) == 0 {
+		return clues.NewWC(ctx, "item added without ID")
+	}
+
 	if parentFolder == nil || len(ptr.Val(parentFolder.GetId())) == 0 {
-		return clues.New("item added without parent folder ID")
+		return clues.NewWC(ctx, "item added without parent folder ID")
 	}
 
 	parentID = ptr.Val(parentFolder.GetId())
 
-	if len(id) == 0 {
-		return clues.New("item added without ID")
-	}
+	ctx = clues.Add(
+		ctx,
+		"parent_id", ptr.Val(parentFolder.GetId()),
+		"parent_dir_path", path.LoggableDir(ptr.Val(parentFolder.GetPath())))
 
 	// in case of file movement, clean up any references
 	// to the file in the old parent
 	oldParentID, ok := face.fileIDToParentID[id]
 	if ok && oldParentID != parentID {
-		if nodey, ok := face.folderIDToNode[oldParentID]; ok {
+		if nodey := face.getNode(oldParentID); nodey != nil {
 			delete(nodey.files, id)
 		}
-
```

Comment:
Is this code removal intentional? Also this change doesn't look related to `rootID` removal, please correct me if wrong.

Going forward, can we please move unrelated changes to a different PR? :)

Reply:
Oh, it's all related at this point.  And, yes, this is intentional.

Reply:
Thanks for clarifying